build.rs

```diff
@@ -0,0 +1,120 @@
+use crate::SqlType::{Idempotent, Migration};
+use askama::Template;
+use std::fs::File;
+use std::io::{Error, Read, Write};
+use std::path::{Path, PathBuf};
+use std::string::String;
+use std::{env, fs};
+
+#[derive(Template)]
+#[template(path = "migration-wrapper.sql", escape = "none")]
+struct MigrationWrapper<'a> {
+    filename: &'a str,
+    version: &'a str,
+    body: String,
+}
+
+#[derive(Template)]
+#[template(path = "idempotent-wrapper.sql", escape = "none")]
+struct IdempotentWrapper<'a> {
+    filename: &'a str,
+    body: String,
+}
+
+const MIGRATION_FILE_NAME: &str = "hand-written-migration.sql";
+
+/// This build script reads the SQL files placed in the `migration/{idempotent,migration}`
+/// directories, wraps each file in its respective template (in the `templates` directory), and
+/// outputs the whole contents as one long SQL file (`MIGRATION_FILE_NAME`) in the root directory.
+fn main() {
+    println!("cargo:rerun-if-changed=Cargo.lock");
+    println!("cargo:rerun-if-changed=migration");
+    println!("cargo:rerun-if-changed=templates");
+    println!("cargo:rerun-if-changed=migration/migration-table.sql");
```

Comment:
Yes, you're right. Looking into it I did find some other changes which needed to be made because of how this actually works. I've also documented it a little better.